scripts/utils/trading_strategy_helpers.js

```diff
@@ -275,14 +329,11 @@ const checkSufficiencyOfBalance = async function(token, owner, amount) {
  * @return {Transaction} Multisend transaction that has to be sent from the master address to either request
 withdrawal of or to withdraw the desired funds
 */
-const buildGenericFundMovement = async function(masterAddress, withdrawals, functionName, web3 = web3, artifacts = artifacts) {
-  BatchExchange.setProvider(web3.currentProvider)
-  BatchExchange.setNetwork(web3.network_id)
-  const exchange = await BatchExchange.deployed()
-  const masterTransactions = []
-
-  // it's not necessary to avoid overlapping withdraws, since the full amount is withdrawn for each entry
-  for (const withdrawal of withdrawals) {
+const buildGenericFundMovement = async function(masterAddress, withdrawals, functionName, web3 = web3, artifacts = artifacts) { // TODO: do we need artifacts here?
+  const exchange = await getExchange(web3)
+  
+  // it's not necessary to avoid overlapping withdraws, since the full amount is withdrawn for each entry TODO: is this still true?
```

Comment:
Yeah, I believe this is still true, as the smart contract logic has not changed:
https://github.com/gnosis/dex-contracts/blob/cc7e166d1ed3430ffd8c06adf757412699fa2d04/contracts/EpochTokenLocker.sol#L111

It might just be more efficient to check for double/overlapping withdraw requests beforehand.

Reply:
True. I also agree it makes sense to check, even just to tell the user that the deposits/withdrawals don't look right.